botocore/loaders.py

```diff
@@ -353,7 +389,7 @@ def load_service_model(self, service_name, type_name, api_version=None):
         return self.load_data(full_path)
 
     @instance_cache
-    def load_data(self, name):
+    def load_data(self, name, use_ordered_dict=True):
```

Comment:
Three ideas that have been juggling here to make it work for the use case I pointed out:
1) Add ``OBJECT_PAIRS_HOOK`` class variable in ``JSONLoader`` to store the class to be used for ``object_pairs_hook``. Downside is that we would have to create a new JSONLoader just for regions and endpoint resolution to avoid affecting loading of service models.
2) Add an extra set of args called ``load_file_params`` that is a dictionary that will be translated to keyword args passed to the underlying ``load_file`` method.
3) We save the use of non-ordered dict to another pull request when we have a better idea of how we want to account for these changes since endpoints are a single file now and load time is not affected as much.

I am not sure if I really like any of these ideas. My preference in order from highest to lowest would be 2, 3, 1. Thoughts?